sdks/java/io/mqtt/src/main/java/org/apache/beam/sdk/io/mqtt/MqttIO.java

```diff
@@ -124,51 +122,90 @@ private MqttIO() {
   @AutoValue
   public abstract static class ConnectionConfiguration implements Serializable {
 
-    @Nullable abstract String serverUri();
-    @Nullable abstract String topic();
-    @Nullable abstract String clientId();
+    @Nullable abstract String getServerUri();
+    @Nullable abstract String getTopic();
+    @Nullable abstract String getClientId();
+    @Nullable abstract String getUsername();
+    @Nullable abstract String getPassword();
 
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setServerUri(String serverUri);
+      abstract Builder setTopic(String topic);
+      abstract Builder setClientId(String clientId);
+      abstract Builder setUsername(String username);
+      abstract Builder setPassword(String password);
+      abstract ConnectionConfiguration build();
+    }
+
+    /**
+     * Describe a connection configuration to the MQTT broker. This method creates an unique random
+     * MQTT client ID.
+     *
+     * @param serverUri The MQTT broker URI.
+     * @param topic The MQTT getTopic pattern.
+     * @return A connection configuration to the MQTT broker.
+     */
     public static ConnectionConfiguration create(String serverUri, String topic) {
       checkArgument(serverUri != null,
-          "MqttIO.ConnectionConfiguration.create(serverUri, topic) called with null "
-              + "serverUri");
+          "MqttIO.ConnectionConfiguration.create(getServerUri, getTopic) called with null "
```

Comment:
`getServerUri` - erroneous global replace? (also above and below - e.g. "getTopic pattern" etc)